src/pages/learn/javascript/use-the-intersection-observer-api-for-analytics-events/index.mdx

```diff
@@ -0,0 +1,307 @@
+
+## Tracking When An Element Comes Into View
+
+The is the basic usage of an Intersection Observer and may be useful for metrics such as which page sections were viewed.
+
+In terms of analytics value - a "viewed" element doesn't mean the user received value from that element, it just means they scrolled by it. You'll want to combine this information with other analytics data such as bounce rate, outbound clicks, and other more specific goal events.
+
+For our example event, we'll attempt to make this milestone meaningful by only tracking the event when the element crosses 50% of the viewport.
+
+To accomplish this, we'll set the `rootMargin` to `0 0 -50% 0`. This means the bounding box of the observable area only extends 50% down the viewport, so the target element must be visible _above_ the halfway point to trigger the callback.
+
+_Note: If you are tracking something at the very start or end of your page, you will not want to adjust the_ `*rootMargin*` _- see the next section for more info_.
+
+**Create An Intersection Observer**
+
+First, we'll setup a check to ensure `IntersectionObserver` is available in the user's browser:
+
+```js
+if ('IntersectionObserver' in window) {
+  // rest of code goes here
+}
+```
+
+For our example, we'd like to detect when an ad becomes visible.
+
+Here, we initiate the observer - `adObserver` - and then provide it the element to observe, an element with the `id` of `ad`:
+
+```js
+// Observer
+const adObserver = new IntersectionObserver()
+// Observation code with go here
+
+// Observation target - `#ad`
+adObserver.observe(document.getElementById('ad'))
+```
+
+The `IntersectionObserver` will receive an array of "entries", even when we are only attaching to one element like in this example.
+
+So, we'll create a variable to hold only the first entry. Then, we'll ensure that it `isIntersecting`, else exit the function (`return`).
+
+If it is intersecting, we can proceed to trigger the analytics event.
+
+We also include an `unobserve` event to ensure our event only triggers once; else it would continue to trigger as the user scrolled up and down past the target element. If you want that repetitive behavior, simply remove that line.
+
+```js
+const adObserver = new IntersectionObserver(
+  (entries) => {
+    const ad = entries[0]
+    if (!ad.isIntersecting) return
+    adObserver.unobserve(ad.target)
+
+    // Tracking code goes here
+  },
+  {
+    rootMargin: '0px 0px -50% 0px',
+  },
+)
+```
+
+You'll then notice that `rootMargin` was added as was discussed earlier into the `options` portion of the `IntersectionObserver`.
+
+Here's a CodePen to help visualize how this `IntersectionObserver` will work:
+`https://codepen.io/5t3ph/live/1a95334057bdf2e5b098932290f1ab62`
```

Comment:
I can move these to CodeSandbox if that would let you embed them? Let me know, otherwise I can make them public on CodePen now to be able to embed as well (if you want to support that embed).

Reply:
I see this is merged, but the CodePens are public now 👍